src/net_processing.cpp

```diff
@@ -2819,6 +2842,20 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             nCMPCTBLOCKVersion = 1;
             m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, fAnnounceUsingCMPCTBLOCK, nCMPCTBLOCKVersion));
         }
+
+        if (m_txreconciliation) {
+            LOCK(cs_main);
+            const auto peer_id = pfrom.GetId();
+            if (!State(peer_id)->m_wtxid_relay) {
+                // We could have optimistically pre-registered the peer on sending SENDRECON, or
+                // registered upon receiving SENDRECON afterwards.
+                // Now we are sure they won't announce WTXIDRELAY (can't happen after VERACK).
+                // We should clear the reconciliation state of the peer (because we support
+                // reconciliations only based on wtxids).
+                m_txreconciliation->ForgetPeer(peer_id);
```

Comment:
Note the BIP still says "Should be ignored if received before receiving "wtxidrelay" from the peer". I think spec should be aligned on code to reflect the reception order dependency has been removed.